service.subtitles.wizdom/service.py

```diff
@@ -1,297 +1,345 @@
-    try:
-        return unquote_plus(params[name])
-    except:
-        pass
-
-
-def GetJson(imdb, season=0, episode=0, version=0):
-    filename = 'wizdom.imdb.%s.%s.%s.json' % (imdb, season, episode)
-    url = "http://api.wizdom.xyz/search.id.php?imdb=%s&season=%s&episode=%s&version=%s" % (
-    imdb, season, episode, version)
-
-    MyLog("GetJson:%s" % url)
-    json_object = Caching(filename, url)
-    subs_rate = []
-    if json_object <> 0:
-        for item_data in json_object:
-            listitem = ListItem(label="Hebrew", label2=item_data["versioname"], thumbnailImage="he",
-                                iconImage="%s" % (item_data["score"] / 2))
-            if int(item_data["score"]) > 8:
-                listitem.setProperty("sync", "true")
-            url = "plugin://%s/?action=download&versioname=%s&id=%s" % (
-            __scriptid__, item_data["versioname"], item_data["id"])
-            addDirectoryItem(handle=int(sys.argv[1]), url=url, listitem=listitem, isFolder=False)
-
-
-def SearchMovie(query, year):
-    filename = 'wizdom.search.movie.%s.%s.json' % (normalizeString(query), year)
-    if year > 0:
-        url = "http://api.tmdb.org/3/search/movie?api_key=f7f51775877e0bb6703520952b3c7840&query=%s&year=%s&language=he" % (
-        quote(query), year)
-    else:
-        url = "http://api.tmdb.org/3/search/movie?api_key=f7f51775877e0bb6703520952b3c7840&query=%s&language=he" % (
-        quote(query))
-    json = Caching(filename, url)
-    try:
-        tmdb_id = int(json["results"][0]["id"])
-    except:
-        return 0
-        pass
-    filename = 'wizdom.tmdb.%s.json' % (tmdb_id)
-    url = "http://api.tmdb.org/3/movie/%s?api_key=f7f51775877e0bb6703520952b3c7840&language=en" % (tmdb_id)
-    json = Caching(filename, url)
-    try:
-        imdb_id = json["imdb_id"]
-    except:
-        return 0
-        pass
-    return imdb_id
-
-
-def Caching(filename, url):
-    json_file = path.join(__temp__, filename)
-    if not path.exists(json_file) or not path.getsize(json_file) > 20 or (time() - path.getmtime(json_file) > 30 * 60):
-        urlretrieve(url, json_file)
-    if path.exists(json_file) and path.getsize(json_file) > 20:
-        with open(json_file) as json_data:
-            json_object = load(json_data)
-        return json_object
-    else:
-        return 0
+	try:
+		return unquote_plus(params[name])
+	except Exception as err:
+		wlog('Caught Exception: error getting param: %s' % format(err))
+		pass
+
+def searchByIMDB(imdb, season=0, episode=0, version=0):
+	filename = 'wizdom.imdb.%s.%s.%s.json' % (imdb, season, episode)
+	url = "http://json.%s/search.php?action=by_id&imdb=%s&season=%s&episode=%s&version=%s" % (
+		myDomain, imdb, season, episode, version)
+
+	wlog("searchByIMDB:%s" % url)
+	json = cachingJSON(filename,url)
+	subs_rate = []  # TODO remove not in used
+	if json != 0:
+		for item_data in json:
+			listitem = ListItem(label="Hebrew", label2=item_data["versioname"],
```

Comment:
drop thumbnailImage and iconImage kwargs. Use only setArt:

`listitem.setArt({'thumb': xxx, 'icon': xxx})`